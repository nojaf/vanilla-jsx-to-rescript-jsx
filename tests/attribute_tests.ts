import { test, expect } from "bun:test";
import transfrom from "../index";

test("type became type_", () => {
  expect(transfrom('<input type="text" />')).toMatchSnapshot();
});

test("single quotes become double", () => {
  expect(transfrom("<input type='text' />")).toMatchSnapshot();
});

test("punning becomes prop=true", () => {
  expect(transfrom("<input disabled />")).toMatchSnapshot();
});

test("open becomes open_", () => {
  expect(transfrom("<dialog open />")).toMatchSnapshot();
});

test("as becomes as", () => {
  expect(transfrom('<dialog as="foo" />')).toMatchSnapshot();
});

test("aria-details becomes ariaDetails", () => {
  expect(transfrom('<dialog aria-details="foo" />')).toMatchSnapshot();
});

test("aria-grabbed becomes ariaGrabbed", () => {
  expect(transfrom("<dialog aria-grapped />")).toMatchSnapshot();
});

test("data-testid becomes dataTestid", () => {
  expect(transfrom('<dialog data-testid="foo" />')).toMatchSnapshot();
});

test("dark prop punning", () => {
  expect(
    transfrom(`<dark-mode-toggle
    id="dark-mode-toggle-1"
    legend="Theme Switcher"
    appearance="switch"
    dark      
    light="Light"
    remember="Remember this"
  ></dark-mode-toggle>`),
  ).toMatchSnapshot();
});
