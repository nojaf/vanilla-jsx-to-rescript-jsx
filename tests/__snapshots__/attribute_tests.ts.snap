// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`type became type_ 1`] = `"<input type_="text" />"`;

exports[`single quotes become double 1`] = `"<input type_="text" />"`;

exports[`punning becomes prop=true 1`] = `"<input disabled=true />"`;

exports[`open becomes open_ 1`] = `"<dialog open_=true />"`;

exports[`as becomes as 1`] = `"<dialog as_="foo" />"`;

exports[`aria-details becomes ariaDetails 1`] = `"<dialog ariaDetails="foo" />"`;

exports[`aria-grabbed becomes ariaGrabbed 1`] = `"<dialog ariaGrapped=true />"`;

exports[`data-testid becomes dataTestid 1`] = `"<dialog dataTestId="foo" />"`;

exports[`dark prop punning 1`] = `
"<dark-mode-toggle
    id="dark-mode-toggle-1"
    legend="Theme Switcher"
    appearance="switch"
    dark=true      
    light="Light"
    remember="Remember this"
  ></dark-mode-toggle>"
`;
